support/cross/README.md

```diff
@@ -0,0 +1,21 @@
+# Cross Compiling
+
+General documentation for [cross-compiling with the Meson build
+system](https://mesonbuild.com/Cross-compilation.html) can be read on their
+site.
+
+Remember to disable 'sanitize' when building for Windows (at some point we may
+figure out how to efficiently add the correct libraries to add ASAN to Windows
+builds).
+
+## x86_64
+
+### Windows
+
+On Linux (or any other system but Windows, since it's a cross-build)
+
+    meson setup --cross-file support/cross/x86_64-w64-mingw32.txt -Db_sanitize=none build-mingw
+
+## aarch64 Linux
+
+    meson setup --cross-file=support/cross/aarch64-linux-gcc.txt build-aarch64
```

Comment:
Oddly enough, on my PC running Manjaro Linux, the Windows build completed, but the aarch Linux build failed. I tried to find a solution but haven't yet. Either @eli-schwartz will magically appear and give me the answer, or I'll put this aside for now and try again later.

```
22/927] Compiling C object subprojects/lua-5.1.5/lib_lua.a.p/src_loslib.c.o
FAILED: subprojects/lua-5.1.5/lib_lua.a.p/src_loslib.c.o 
aarch64-linux-gnu-gcc -Isubprojects/lua-5.1.5/lib_lua.a.p -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c99 -O0 -g -Wno-stringop-overflow -DLUA_USE_POSIX -fPIC -MD -MQ subprojects/lua-5.1.5/lib_lua.a.p/src_loslib.c.o -MF subprojects/lua-5.1.5/lib_lua.a.p/src_loslib.c.o.d -o subprojects/lua-5.1.5/lib_lua.a.p/src_loslib.c.o -c ../subprojects/lua-5.1.5/src/loslib.c
In file included from ../subprojects/lua-5.1.5/src/lua.h:16,
                 from ../subprojects/lua-5.1.5/src/loslib.c:17:
../subprojects/lua-5.1.5/src/loslib.c: In function 'os_tmpname':
../subprojects/lua-5.1.5/src/luaconf.h:651:13: error: implicit declaration of function 'mkstemp' [-Wimplicit-function-declaration]
  651 |         e = mkstemp(b); \
      |             ^~~~~~~
../subprojects/lua-5.1.5/src/loslib.c:60:3: note: in expansion of macro 'lua_tmpnam'
   60 |   lua_tmpnam(buff, err);
      |   ^~~~~~~~~~
[24/927] Compiling C object subprojects/lua-5.1.5/lib_lua.a.p/src_liolib.c.o
FAILED: subprojects/lua-5.1.5/lib_lua.a.p/src_liolib.c.o 
aarch64-linux-gnu-gcc -Isubprojects/lua-5.1.5/lib_lua.a.p -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c99 -O0 -g -Wno-stringop-overflow -DLUA_USE_POSIX -fPIC -MD -MQ subprojects/lua-5.1.5/lib_lua.a.p/src_liolib.c.o -MF subprojects/lua-5.1.5/lib_lua.a.p/src_liolib.c.o.d -o subprojects/lua-5.1.5/lib_lua.a.p/src_liolib.c.o -c ../subprojects/lua-5.1.5/src/liolib.c
In file included from ../subprojects/lua-5.1.5/src/lua.h:16,
                 from ../subprojects/lua-5.1.5/src/liolib.c:16:
../subprojects/lua-5.1.5/src/liolib.c: In function 'io_pclose':
../subprojects/lua-5.1.5/src/luaconf.h:671:44: error: implicit declaration of function 'pclose'; did you mean 'fclose'? [-Wimplicit-function-declaration]
  671 | #define lua_pclose(L,file)      ((void)L, (pclose(file) != -1))
      |                                            ^~~~~~
../subprojects/lua-5.1.5/src/liolib.c:110:12: note: in expansion of macro 'lua_pclose'
  110 |   int ok = lua_pclose(L, *p);
      |            ^~~~~~~~~~
../subprojects/lua-5.1.5/src/liolib.c: In function 'io_popen':
../subprojects/lua-5.1.5/src/luaconf.h:670:57: error: implicit declaration of function 'popen'; did you mean 'fopen'? [-Wimplicit-function-declaration]
  670 | #define lua_popen(L,c,m)        ((void)L, fflush(NULL), popen(c,m))
      |                                                         ^~~~~
../subprojects/lua-5.1.5/src/liolib.c:178:9: note: in expansion of macro 'lua_popen'
  178 |   *pf = lua_popen(L, filename, mode);
      |         ^~~~~~~~~
../subprojects/lua-5.1.5/src/liolib.c:178:7: error: assignment to 'FILE *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  178 |   *pf = lua_popen(L, filename, mode);
```

Reply:
This is a bunch of errors that all happen when you use GCC 14 (cross compilation not relevant) due to latent UB because the source code is attempting to include headers and then use functions which do not actually exist.

It's actually fixed by upgrading to a new lua version. That new lua version checks if _XOPEN_SOURCE is defined, and if not, sets it to 600.

Reply:
Thank you @eli-schwartz